lib/executor/variable_looping_vus_test.go

```diff
@@ -1002,3 +1004,117 @@ func TestSegmentedIndex(t *testing.T) {
 		assert.EqualValues(t, 0, s.scaled)
 	})
 }
+
+// TODO: delete in favor of lib.generateRandomSequence() after
+// https://github.com/loadimpact/k6/issues/1302 is done (can't import now due to
+// import loops...)
+func generateRandomSequence(t testing.TB, n, m int64, r *rand.Rand) lib.ExecutionSegmentSequence {
+	var err error
+	ess := lib.ExecutionSegmentSequence(make([]*lib.ExecutionSegment, n))
+	numerators := make([]int64, n)
+	var denominator int64
+	for i := int64(0); i < n; i++ {
+		numerators[i] = 1 + r.Int63n(m)
+		denominator += numerators[i]
+	}
+	from := big.NewRat(0, 1)
+	for i := int64(0); i < n; i++ {
+		to := new(big.Rat).Add(big.NewRat(numerators[i], denominator), from)
+		ess[i], err = lib.NewExecutionSegment(from, to)
+		require.NoError(t, err)
+		from = to
+	}
+
+	return ess
+}
+
+func TestSumRandomSegmentSequenceMatchesNoSegment(t *testing.T) {
+	t.Parallel()
+
+	seed := time.Now().UnixNano()
+	r := rand.New(rand.NewSource(seed))
+	t.Logf("Random source seeded with %d\n", seed)
+
+	const (
+		numTests         = 10
+		maxStages        = 10
+		minStageDuration = 1 * time.Second
+		maxStageDuration = 10 * time.Minute
+		maxVUs           = 300
+		segmentSeqMaxLen = 15
+		maxNumerator     = 300
+	)
+	getTestConfig := func(name string) VariableLoopingVUsConfig {
+		stagesCount := 1 + r.Int31n(maxStages)
+		stages := make([]Stage, stagesCount)
+		for s := int32(0); s < stagesCount; s++ {
+			dur := time.Duration(r.Int63n(int64(maxStageDuration - minStageDuration))).Round(time.Second)
+			stages[s] = Stage{Duration: types.NullDurationFrom(dur), Target: null.IntFrom(r.Int63n(maxVUs))}
+		}
+
+		c := NewVariableLoopingVUsConfig(name)
+		c.GracefulRampDown = types.NullDurationFrom(0)
+		c.GracefulStop = types.NullDurationFrom(0)
+		c.StartVUs = null.IntFrom(r.Int63n(maxVUs))
+		c.Stages = stages
+		return c
+	}
+
+	subtractChildSteps := func(t *testing.T, parent, child []lib.ExecutionStep) {
+		t.Logf("subtractChildSteps()")
+		for _, step := range child {
+			t.Logf("	child planned VUs for time offset %s: %d", step.TimeOffset, step.PlannedVUs)
+		}
+		sub := uint64(0)
+		ci := 0
+		for pi, p := range parent {
+			// We iterate over all parent steps and match them to child steps.
+			// Once we have a match, we remove the child step's plannedVUs from
+			// the parent steps until a new match, when we adjust the subtracted
+			// amount again.
+			if p.TimeOffset > child[ci].TimeOffset && ci != len(child)-1 {
+				t.Errorf("ERR Could not match child offset %s with any parent time offset", child[ci].TimeOffset)
+			}
+			if p.TimeOffset == child[ci].TimeOffset {
+				t.Logf("Setting sub to %d at t=%s", child[ci].PlannedVUs, child[ci].TimeOffset)
+				sub = child[ci].PlannedVUs
+				if ci != len(child)-1 {
+					ci++
+				}
+			}
+			t.Logf("Subtracting %d VUs (out of %d) at t=%s", sub, p.PlannedVUs, p.TimeOffset)
```

Comment:
This test is very noisy when run with `-v`, which I usually enable to see the full output, but now this test prints ~80K lines, forcing me to use `tee`, etc. :-/

Can we use a debug logger here?

Reply:
I am for commenting all the `Logf` lines and leaving only the random seed one as that is needed to get it reproduced - if you get an error on this, trust me you would want to be able to reproduce it ... the size of the output is way too big to just read it in circleci for example